### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by Jorbreezy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,45 @@
-# Project 1
+# Reddit Posts Search
```
Comment (joelcolucci) [inline README.md:1]: Nice work improving this!

File: lib/index.js [hunk 2]
```diff
@@ -1,8 +1,76 @@
...
+   // In process data do we want it allow the developer to choose what 
+
+
+export const fetchData = (url) => {
+  return fetch(url)
+    .then(res => res.json())
+    .catch(err => console.error(err))
+}
+
+export const processData = (res, fields) => {
```
Comment (joelcolucci) [inline lib/index.js:22]: Nice job consolidating knowledge of the external data model to this function!

File: lib/index.js [hunk 3]
```diff
@@ -1,8 +1,76 @@
...
+}
+
+export const getRedditUrl = (topic, sort = "new") => {
+  if (!topic && typeof topic !== 'string') return new Error('Topic not a string!');
+  if (typeof sort !== 'string') return new Error('Sort is not a string!'); 
+
+  return `https://www.reddit.com/r/pics/search.json?q=${topic}&sort=${sort}`;
+}
+
+export const saveResults = (filename, data) => {
```
Comment (joelcolucci) [inline lib/index.js]: I like that we factored this out. Nice work!

### What is this functions single responsibility?
It appears the single responsibility is to write a JSON file

### Question
Can we rename this function to match it's single responsibility?

File: lib/index.js [hunk 4]
```diff
@@ -1,8 +1,76 @@
...
+}
+
+// Pass all the data into specify fields
+   // Extract Fields
+   // Allow it to be specific to an array of objects
+   // Could add a function Process data and use specify/extract fields in it
+   // In process data do we want it allow the developer to choose what 
+
+
+export const fetchData = (url) => {
```
Comment (joelcolucci) [inline lib/index.js]: Nice job, factoring this out!

### What is this functions single responsibility?
The single responsibility is to fetch json data

### Question
Can we rename this function to match it's single responsibility?

File: package.json [hunk 5]
```diff
@@ -4,21 +4,24 @@
...
     "dev": "nodemon --exec babel-node lib/index.js",
     "build": "babel lib --out-dir build",
     "start": "node build/index.js"
   },
   "author": "Jordan Kelly",
   "license": "ISC",
   "dependencies": {
-    "node-fetch": "^2.6.1"
+    "node-fetch": "^2.6.1",
+    "path": "^0.12.7"
```
Comment (joelcolucci) [inline package.json]: `path` is a standard node library. We should be able to include it, without it being in our deps.

File: test/index.test.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+const { searchRedditPostsByTopicPromise } = require('../build/index');
```
Comment (joelcolucci) [inline test/index.test.js]: Our imports should be organized into the following sections:
1. standard node libraries
2. external npm libraries
3. our own imports

Each section should be alphabetized.

### Why?
Standardizing organization makes it easier for developers to read and understand a project.

In addition, when collaborating with other developers on large projects, merge conflicts can pop up in the import section if there's not a pattern to follow.

File: tests/index.test.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    it('Should match returned data', async () => {
+      const redditUrl = 'https://reddit.com/r/pics/search.json?q=Gaming&sort=new';
+
+      const responseData = { data: { children: [{ title: 'RTX 3080 Crashes' }] } };
+
+      nock('https://reddit.com')
+        .get('/r/pics/search.json?q=Gaming&sort=new')
+        .reply(200, responseData);
+
+      return await fetchJson(redditUrl)
```
Comment (joelcolucci) [inline tests/index.test.js]: Let's clean up this syntax. Currently it's a mix of async/await and a .then promise chain.

Let's just use async/await similar to test below.

File: tests/index.test.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+      const request = await fetchRedditPostsByTopic('Gaming');
+
+      expect(request).toEqual(children);
+    });
+
+    it('Should return a certain response when topic and fields is passed', async () => {
+      const fields = ['id', 'title'];
+
+      const { data: { children } } = response;
```
Comment (joelcolucci) [inline tests/index.test.js]: Let's rename response. Response indicates it is an HTTP response object or something similar.

In our case it's a mock response. Let's name it `mockResponse` to make it clear if we are dealing with a mock or "actual" response.

File: tests/index.test.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+      nock('https://www.reddit.com')
+        .get('/r/pics/search.json?q=Gaming&sort=new')
+        .reply(200, response);
+
+      const request = await fetchRedditPostsByTopic('Gaming', fields);
+
+      const expectedResponse = extractFields(children, fields);
```
Comment (joelcolucci) [inline tests/index.test.js]: Are we testing extract fields here or fetchRedditPostsByTopic? If fetchRedditPostsByTopic we should not rely on extractFields. Instead creating another mock object to compare against.

Something like `mockFetchRedditPostByTopicResponse`

Comment (Jorbreezy) [reply tests/index.test.js]: I used extractFields just so I don't have to create the mock object since it would be the same thing.

**Review by joelcolucci — APPROVED** [review]
Nice work! It's really coming together 💪

**Pull request merged**